### PR TITLE
Keep a list of operations that needs to be updated in the database

### PR DESCRIPTION
**Pull request by tcezard**

### CODE REVIEW

File: eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java [hunk 1]
```diff
@@ -318,6 +323,7 @@ private boolean checkIfExistingRSOrCandidateForMerge(SubmittedVariantEntity subm
                 submittedVariantOperationEntity.fill(EventType.RS_MERGE_CANDIDATES, accessionInDB,
                         "RS mismatch with " + accessionInDB, inactiveObjects);
 
+                updateMergeCandidateSVOE.put(variantHash, submittedVariantOperationEntity);
```
Comment (sundarvenkata-EBI) [inline eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java:326]: This map should be populated [here](https://github.com/EBIvariation/eva-accession/pull/315/files#diff-b6244e9907d2b0296d3a02a3fc6481f3fb8656821add15fea30f85a243ce3c59R314) too..

File: eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java [hunk 2]
```diff
@@ -355,6 +362,7 @@ private void checkIfCandidateForRSSplit(SubmittedVariantEntity submittedVariantE
                         "Hash mismatch with " + variantAccession, inactiveEntities);
 
                 rsSplitCandidateSVOE.put(variantAccession, submittedVariantOperationEntity);
+                updateRsSplitCandidateSVOE.put(variantAccession, submittedVariantOperationEntity);
```
Comment (sundarvenkata-EBI) [inline eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java:365]: This map should be populated [here](https://github.com/EBIvariation/eva-accession/blob/614bdfe8a91dccdffec1195738f9225805d3a4cf/eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java#L352) too...

**Review by sundarvenkata-EBI — APPROVED** [review]

File: eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java [hunk 3]
```diff
@@ -305,8 +310,10 @@ private boolean checkIfExistingRSOrCandidateForMerge(SubmittedVariantEntity subm
 
         if (allExistingHashesInDB.containsKey(variantHash)) {
             if (mergeCandidateSVOE.containsKey(variantHash)) {
-                mergeCandidateSVOE.get(variantHash).getInactiveObjects()
+                SubmittedVariantOperationEntity submittedVariantOperationEntity = mergeCandidateSVOE.get(variantHash);
+                submittedVariantOperationEntity.getInactiveObjects()
                         .add(new SubmittedVariantInactiveEntity(submittedVariantEntity));
+                updateMergeCandidateSVOE.put(variantHash, submittedVariantOperationEntity);
```
Comment (andresfsilva) [inline eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java:316]: This is present in the if and else statements, can be taken out so it is not repeated

File: eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java [hunk 4]
```diff
@@ -326,25 +334,27 @@ private boolean checkIfExistingRSOrCandidateForMerge(SubmittedVariantEntity subm
...
+                SubmittedVariantOperationEntity submittedVariantOperationEntity = rsSplitCandidateSVOE.get(variantAccession);
+                List<SubmittedVariantInactiveEntity> inactiveEntities = submittedVariantOperationEntity.getInactiveObjects();
                 boolean submittedVariantAlreadyExist = inactiveEntities.stream()
                         .anyMatch(sv -> submittedHashingFunction.apply(sv).equals(
                                 submittedVariantEntity.getHashedMessage()) &&
                                 sv.getAccession().equals(submittedVariantEntity.getAccession()));
                 if (!submittedVariantAlreadyExist) {
                     inactiveEntities.add(new SubmittedVariantInactiveEntity(submittedVariantEntity));
                 }
+                updateRsSplitCandidateSVOE.put(variantAccession, submittedVariantOperationEntity);
```
Comment (andresfsilva) [inline eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java:357]: This is present in the if and else statements, can be taken out so it is not repeated

Comment (tcezard) [reply eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java:359]: Done

**Pull request merged**